pkg/actions/cluster/unowned.go

```diff
@@ -0,0 +1,40 @@
+package cluster
+
+import (
+	"github.com/kris-nova/logger"
+	api "github.com/weaveworks/eksctl/pkg/apis/eksctl.io/v1alpha5"
+	"github.com/weaveworks/eksctl/pkg/eks"
+	"github.com/weaveworks/eksctl/pkg/printers"
+)
+
+type UnownedCluster struct {
+	cfg *api.ClusterConfig
+	ctl *eks.ClusterProvider
+}
+
+func NewUnownedCluster(cfg *api.ClusterConfig, ctl *eks.ClusterProvider) (*UnownedCluster, error) {
+	return &UnownedCluster{
+		cfg: cfg,
+		ctl: ctl,
+	}, nil
+}
+
+func (c *UnownedCluster) Upgrade(dryRun bool) error {
```

Comment:
this function seems to be duplicated in `owned.go` except for the `LoadClusterVPC` call, can that be reduced somehow?

Reply:
lmk what you think @michaelbeaumont https://github.com/weaveworks/eksctl/pull/2815/commits/a6e01da33ca345697881b96da1307b538e61b92f